exonum-java-binding-core/src/main/java/com/exonum/binding/service/Node.java

```diff
@@ -28,19 +28,22 @@
 public interface Node {
 
   /**
-   * Submits a transaction into Exonum network. This node does <em>not</em> execute
-   * the transaction immediately, but verifies it and, if it is valid,
-   * broadcasts it to all the nodes in the network. Then each node adds the transaction to a
+   * Creates a transaction from the given parameters and sign it with its service key.
+   * And then submits it into Exonum network. This node does <em>not</em> execute
+   * the transaction immediately, broadcasts it to all the nodes in the network.
+   * Then each node adds the transaction to a
    * <a href="https://exonum.com/doc/advanced/consensus/specification/#pool-of-unconfirmed-transactions">pool of unconfirmed transactions</a>.
    * The transaction is executed later asynchronously.
+   * <em>Be aware that invocations of this method on different nodes
+   * will produce different transactions.</em>
    *
-   * @param transaction a transaction to send
-   * @throws InvalidTransactionException if the transaction is not valid
+   * @param rawTransaction transaction parameters to include in transaction message
+   * @return hash of the transaction message created by the framework
    * @throws InternalServerError if this node failed to process the transaction
    * @throws NullPointerException if the transaction is null
    */
-  void submitTransaction(Transaction transaction)
-      throws InvalidTransactionException, InternalServerError;
+  byte[] submitTransaction(RawTransaction rawTransaction)
```

Comment:
HashCode. But keep the method below with `byte[]` as it involves a little more, see https://jira.bf.local/browse/ECR-2758